server/tests/test_browser_viewer.py

```
@@ -0,0 +1,545 @@
+    assert voted_mark.count() == 1
+    mark_text = voted_mark.text_content()
+    assert "已投出" in mark_text or "Voted" in mark_text
+
+    # Product rule: never render vote counts / percentages anywhere on the
+    # viewer poll pane.
+    stat_elements = page.locator(".viewer-poll-option-stat")
+    assert stat_elements.count() == 0, "Viewer must never show vote counts/percentages"
+
+
+def test_poll_pane_never_shows_percentage_text(viewer_page, admin_http):
+    """即使投票有多筆，viewer 的 poll pane 文字內容也不應含百分比符號"""
```

Ruff (0.15.21)

[warning] Docstring contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF002)